### ISSUE TITLE
[Feature]: A way to disable soft FoW / visibility in night maps

### ISSUE THREAD

**Issue by kwvanderlinde** [RPTools/maptool#4232]
### Describe the Problem

I find I don't really need / care about soft FoW in my games, and it sometimes causes confusion as players don't always see the tokens I expect. At the same time, we like playing on night maps, with lights, sights, and hard FoW being part of the experience. But right now, I can only disable soft FoW by setting vision to "Off", which also makes the map behave like a day map.

### The Solution you'd like

I would like to split the map vision option into two separate concepts (names TBD):
- **Visibility**: a toggle that determines whether to render soft FoW, hide tokens out of line-of-sight, etc. I.e., controls the things that are currently disabled when switching from "Day" to "Off".
- **Lighting**: selection of "Day" or "Night" that determines whether lights (personal or otherwise) are required to illuminate areas, or whether the entire vision range is considered illuminated.

### Additional Context

This is a simpler version of #3822 that just focuses on separating the "Off" functionality from the others.

**Comment by FullBleed** [comment]
Would just having the ability to toggle Soft Fog on/off achieve what you want?

**Comment by kwvanderlinde** [comment]
That's essentially what the **Visibility** toggle is meant to be (I've now updated the description to call out soft FoW specifically). The various behaviours there are collectively what "soft FoW" is.

I guess calling it the **Soft FoW** toggle would make more sense in that case ...

**Comment by kwvanderlinde** [comment]
There is one more behaviour of the **Vision** setting that came up on Discord which I was surprised to find out about (though I should have known considering the implementation). When **Vision** is set to "Off", the server setting of **Individual FoW** is kind of ignored (though not totally): any fog exposure becomes global to the map in addition to being exposed for each token.

This is a behaviour I would _not_ want controlled by the new **Visibility** / **Soft FoW** options, and certainly not by the new **Lighting** option. Actually, I would rather the behaviour didn't exist, but to support existing use cases I will instead propose that it be made a third new map setting, a toggle along the lines of **Always Expose FoW Globally**.

**Comment by FullBleed** [comment]
It feels like the current behaviour is kind of a bug... and I think I'd find the "Always Expose FoW Globally" toggle to be confusing (especially for new users trying to understand how everything interacts.)

Would an "Always Expose FoW Globally" toggle, essentially, be an Individual FoW toggle?

**Comment by kwvanderlinde** [comment]
> It feels like the current behaviour is kind of a bug...

I would agree. Or rather I should say I don't think the current behaviour is desirable. After digging into the history a bit, it's clear the choice was deliberate for supporting maps that are just meant for temporarily parking tokens (see this [forum thread](https://forums.rptools.net/viewtopic.php?t=17228), and especially [this post](https://forums.rptools.net/viewtopic.php?p=183592#p183592)). That use case is fine IMO, I would just rather it be called out as its own thing or be somehow incorporated into the fog settings instead of being attached to the otherwise unrelated Vision settings.

> and I think I'd find the "Always Expose FoW Globally" toggle to be confusing (especially for new users trying to understand how everything interacts.)

I won't argue with that :slightly_smiling_face: But as mentioned above, there is a use case here, so _some_ option for it is necessary. Of course it doesn't at all have to look like what I suggested, I'm sure there's a much more intuitive alternative.

> Would an "Always Expose FoW Globally" toggle, essentially, be an Individual FoW toggle?

I think it could be, but depends how attached we are to the details of the current behaviour. Technically the current behaviour is that Vision=Off can act like Individual FoW and Global FoW are _both_ enabled (assuming the Individual FoW server setting is applied). That means it could have an impact if the map/server settings change over time, though any reliance on that seems like quite an edge case to me.

**Comment by kwvanderlinde** [comment]
Another thing that routinely gets asked about is how to restore hard FoW as tokens move around. I think this would be a natural extension of the **Visibility** option described here. So it would end up with options like:
- Off (ignore vision altogether)
- Use only hard FoW, with history (keep exposed areas exposed)
- Use only hard FoW, without history (keep only currently visible areas exposed)
- Use soft FoW + hard FoW

**Comment by FullBleed** [comment]
Would this be accurate?

Off (Ignore Vision)
Fog of War with History
--Line of Sight
----Use Soft FoW
Fog of War without History

With "LoS" and "Soft FoW" each being optional and additive to "FoW with History"?
